protocol/src/main/java/org/geysermc/mcprotocollib/protocol/MinecraftProtocol.java

```diff
@@ -186,17 +201,35 @@ protected PacketEncryption enableEncryption(Key key) {
     }
 
     /**
-     * Gets the current {@link ProtocolState} the client is in.
+     * Gets the current inbound {@link ProtocolState} we're in.
      *
-     * @return The current {@link ProtocolState}.
+     * @return The current inbound {@link ProtocolState}.
      */
-    public ProtocolState getState() {
-        return this.state;
+    public ProtocolState getInboundState() {
+        return this.inboundState;
     }
 
-    public void setState(ProtocolState state) {
-        this.state = state;
-        this.stateRegistry = this.codec.getCodec(state);
+    /**
+     * Gets the current outbound {@link ProtocolState} we're in.
+     *
+     * @return The current outbound {@link ProtocolState}.
+     */
+    public ProtocolState getOutboundState() {
+        return this.outboundState;
+    }
+
+    public void setInboundState(ProtocolState state) {
+        log.debug("Setting inbound state to: {}", state);
```

Comment:
```suggestion
        log.debug("Setting inbound protocol state to: {}", state);
```
Might be just a bit clearer